src/surroundings_menu.cpp

```diff
@@ -0,0 +1,1160 @@
+    if( switch_tab == surroundings_menu_tab_enum::monsters ) {
+        flags = ImGuiTabItemFlags_SetSelected;
+        switch_tab = surroundings_menu_tab_enum::num_tabs;
+    }
+    if( ImGui::BeginTabItem( monster_data.title.c_str(), nullptr, flags ) ) {
+        selected_tab = surroundings_menu_tab_enum::monsters;
+        if( ImGui::BeginTable( "monsters", 5, ImGuiTableFlags_ScrollY,
+                               ImVec2( 0.0f, str_height_to_pixels( TERMY - info_height ) ) ) ) {
+            ImGui::TableSetupColumn( "mon_sees", ImGuiTableColumnFlags_NoDirectResize_,
+                                     str_width_to_pixels( 1 ) );
+            ImGui::TableSetupColumn( "mon_name", ImGuiTableColumnFlags_NoDirectResize_,
+                                     str_width_to_pixels( width - dist_width - 26 ) );
+            ImGui::TableSetupColumn( "mon_health", ImGuiTableColumnFlags_NoDirectResize_,
+                                     str_width_to_pixels( 5 ) );
+            ImGui::TableSetupColumn( "mon_attitude", ImGuiTableColumnFlags_NoDirectResize_,
+                                     str_width_to_pixels( 18 ) );
+            ImGui::TableSetupColumn( "mon_distance", ImGuiTableColumnFlags_NoDirectResize_,
+                                     str_width_to_pixels( dist_width ) );
+            if( !monster_data.selected_entry ||
+                std::find( monster_data.filtered_list.begin(), monster_data.filtered_list.end(),
+                           monster_data.selected_entry ) == monster_data.filtered_list.end() ) {
+                monster_data.selected_entry = monster_data.filtered_list.front();
+            }
+            std::string last_category;
+            int entry_no = 0;
+            for( map_entity_stack<Creature> *it : monster_data.filtered_list ) {
+                if( monster_data.draw_categories ) {
+                    draw_category_separator( it->get_category(), last_category, 1 );
+                }
+                bool is_selected = it == monster_data.selected_entry;
+                ImGui::TableNextColumn();
+                ImGui::PushID( entry_no++ );
+                ImGui::Selectable( "", &is_selected,
+                                   ImGuiSelectableFlags_AllowItemOverlap | ImGuiSelectableFlags_SpanAllColumns, ImVec2( 0, 0 ) );
+                if( is_selected ) {
+                    if( auto_scroll ) {
+                        ImGui::SetScrollHereY();
+                        auto_scroll = false;
+                    }
+                    monster_data.selected_entry = it;
+                }
+                ImGui::SameLine( 0, 0 );
+                ImGui::PopID();
+                const Creature *mon = it->get_selected_entity();
+                const Character &you = get_player_character();
+                const bool inattentive = you.has_trait( trait_INATTENTIVE );
+                bool sees = mon->sees( you ) && !inattentive;
+                cataimgui::draw_colored_text( sees ? "!" : " ", c_yellow );
```

Comment:
When possible, prefer to just call `ImGui::TextColored` instead, as it does less work than `cataimuig::draw_colored_text`.